src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/DocFinder.java

```diff
@@ -47,45 +44,34 @@ public class DocFinder {
         Result<T> apply(ExecutableElement method) throws X;
     }
 
-    private final Function<ExecutableElement, ExecutableElement> overriddenMethodLookup;
-    private final BiFunction<ExecutableElement, ExecutableElement, Iterable<ExecutableElement>> implementedMethodsLookup;
+    private final Function<ExecutableElement, Iterable<? extends ExecutableElement>> overriddenMethodLookup;
 
-    DocFinder(Function<ExecutableElement, ExecutableElement> overriddenMethodLookup,
-              BiFunction<ExecutableElement, ExecutableElement, Iterable<ExecutableElement>> implementedMethodsLookup) {
+    DocFinder(Function<ExecutableElement, Iterable<? extends ExecutableElement>> overriddenMethodLookup) {
         this.overriddenMethodLookup = overriddenMethodLookup;
-        this.implementedMethodsLookup = implementedMethodsLookup;
     }
 
     @SuppressWarnings("serial")
-    public static final class NoOverriddenMethodsFound extends Exception {
+    public static final class NoOverriddenMethodFound extends Exception {
```

Comment:
General comment, for future work:
Maybe we can combine/merge/unify this exception with `ThrowsTaglet.Failure`